source/03-components/modal/modal.es6.js

```diff
@@ -0,0 +1,112 @@
+import Drupal from 'drupal';
+import once from 'once';
+import KEYCODE from '../../00-config/_KEYCODE.es6';
+
+Drupal.behaviors.accordion = {
+  attach(context) {
+    const MODAL_CLASS = 'js-modal';
+    const MODAL_INNER_CLASS = 'js-modal-inner';
+    const MODAL_OPEN_BUTTON_CLASS = 'js-modal-open';
+    const MODAL_CLOSE_BUTTON_CLASS = 'js-modal-close';
+
+    const modals = once('modal', `.${MODAL_CLASS}`, context);
+    const modalOpenButtons = once(
+      'show-modal-button',
+      `.${MODAL_OPEN_BUTTON_CLASS}`,
+      context
+    );
+    const modalCloseButtons = once(
+      'close-modal-button',
+      `.${MODAL_CLOSE_BUTTON_CLASS}`,
+      context
+    );
+
+    // Function to show modal
+    const openModal = modal => {
+      modal.showModal();
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      modal.addEventListener('click', handleOverlayClick);
+      // Turn off scrolling on the body
+      document.body.style.overflow = 'hidden';
+    };
+
+    // Function to close modal
+    const closeModal = modal => {
+      modal.close();
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      modal.removeEventListener('click', handleOverlayClick);
+      // Turn on scrolling on the body
+      document.body.style.overflow = '';
+    };
+
+    // Close modal on overlay click
+    const handleOverlayClick = event => {
+      if (event.target.closest(`.${MODAL_INNER_CLASS}`)) return;
+      closeModal(event.currentTarget);
+    };
```

Comment:
We should add an event here that other scripts can attach listeners to. Something like:

```js
const event = new CustomEvent('closemodal');
modal.dispatchEvent(event);
```